rust/connlib/tunnel/src/client.rs

```diff
@@ -926,7 +1003,59 @@ impl ClientState {
         self.drain_node_events();
 
         self.mangled_dns_queries.retain(|_, exp| now < *exp);
-        self.forwarded_dns_queries.retain(|_, (_, exp)| now < *exp);
+
+        self.tcp_dns_server.handle_timeout(now);
+
+        // Check if have any pending TCP DNS queries.
+        while let Some(query) = self.tcp_dns_server.poll_queries() {
```

Comment:
I only see this called here but I don't see any `poll_timeout` for `tcp_dns_server` and this seem the only way to harvest the TCP dns queries, does this mean we only handle the tcp queries on timeouts depending on other components?

Reply:
It is a bit subtle but `Tunnel::poll` will call `handle_timeout` every time it feeds a packet to the state that **doesn't** result in a `Transmit`. Hence, this is effectively called every time we dispatch a packet to the TCP server.

Reply:
Ah! That makes sense, can we make it a bit more explicit by separating this into a different function? Or we also need it called each time `handle_timeout` happens for other reasons?

Reply:
You mean just extracting a private function? I'd like to keep the call-chain tied to `handle_timeout`. I think any form of background-processing should happen there. That is how our current sans-IO designs are laid out :)

It gets pretty messy if you start assume, where things change in result to other actions.

Open to other ideas though!